tools/ci_connector_ops/ci_connector_ops/pipelines/hacks.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+"""This module contains hacks used in connectors pipelines. They're gathered here for tech debt visibility."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import TYPE_CHECKING
+
+import yaml
+from ci_connector_ops.utils import ConnectorLanguage
+from dagger import DaggerError
+
+if TYPE_CHECKING:
+    from ci_connector_ops.pipelines.contexts import ConnectorContext
+    from dagger import Directory
+
+
+async def _patch_dockerfile(context: ConnectorContext, connector_dir: Directory) -> Directory:
+    """Patch a python connector dockerfile to pin Cython to <3.0 and pyyaml to ~=5.4
+    This patching logic should be removed once we discard Dockerfiles for python connectors in favor of fully daggerized connector build.
+
+    Args:
+        context (ConnectorContext): The initialized connector context.
+        connector_dir (Directory): The directory containing the Dockerfile to patch.
+    Returns:
+        Directory: The directory containing the patched dockerfile.
+    """
+
+    if context.connector.language not in [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]:
+        context.logger.info(f"Connector language {context.connector.language} does not require a patched dockerfile.")
+        return connector_dir
+    try:
+        dockerfile_content = await connector_dir.file("Dockerfile").contents()
+    except DaggerError:
+        context.logger.warn(f"Could not find Dockerfile in {connector_dir}. Skipping patching.")
+        return connector_dir
+
+    context.logger.warn("Patching dockerfile to pin Cython to <3.0 and pyyaml to ~=5.4")
+    patched_dockerfile = []
+    pinned = False
+    for line in dockerfile_content.splitlines():
+        # Workaround for https://github.com/yaml/pyyaml/issues/601
+        # Cython 3.0 is incompatible with pyyaml ~=5.4, we shall pin Cython to <3.0
+        # Pyyaml should release a new patch version that explicitly pins Cython to <3.0
+        if "RUN pip install" in line and not pinned:
+            patched_dockerfile.append('RUN pip install --prefix=/install "Cython<3.0" "pyyaml~=5.4" --no-build-isolation')
+            patched_dockerfile.append('RUN pip install "Cython<3.0" "pyyaml~=5.4" --no-build-isolation')
+            pinned = True
```

Comment:
Lets emit a log warning when we do this.

If we forget to remove this, we could accidentally cause a bug confusing the heck out of a dev